packages/facility-server/__tests__/apiv1/User.test.js

```diff
@@ -340,108 +355,144 @@ describe('User', () => {
+      mockLoginFacilityPermission(linkedUser, false);
+    });
+
+    beforeAll(async () => {
       superUser = await models.User.create(
         createUser({
           role: 'admin',
         }),
       );
-      user = await models.User.create(
+
+      linkedUser = await models.User.create(
         createUser({
           role: 'practitioner',
         }),
       );
-      noFacilityUser = await models.User.create(
+
+      unlinkedUser = await models.User.create(
         createUser({
           role: 'practitioner',
         }),
       );
 
       await Promise.all(
-        validUserFacilities.map(async (facility) => {
+        linkedUserFacilities.map(async facility => {
           return await models.UserFacility.create({
             facilityId: facility.id,
-            userId: user.id,
+            userId: linkedUser.id,
           });
         }),
       );
 
-      await user.reload({ include: 'facilities' });
-    });
-
-    describe('checkCanAccessAllFacilities', () => {
-      it('should return true if superuser', async () => {
-        expect(await superUser.checkCanAccessAllFacilities()).toBe(true);
-      });
-      it('should return false if user without  "login", "Facility" permission', async () => {
-        expect(await user.checkCanAccessAllFacilities()).toBe(false);
-      });
+      await linkedUser.reload({ include: 'facilities' });
     });
 
     describe('allowedFacilities', () => {
       it('should get special "ALL" key when superuser', async () => {
         const superUserFacilities = await superUser.allowedFacilities();
         expect(superUserFacilities).toBe(CAN_ACCESS_ALL_FACILITIES);
       });
+
+      it('user with login permission should return all non-sensitive facilities plus linked facilities', async () => {
+        mockLoginFacilityPermission(linkedUser, true);
+        const userFacilities = await linkedUser.allowedFacilities();
+        expect(userFacilities).toEqual(expect.arrayContaining(expectedCombinedFacilities));
+      });
+
+      it('setting disabled should return all non-sensitive facilities plus linked facilities', async () => {
+        await models.Setting.set('auth.restrictUsersToFacilities', false);
+        const userFacilities = await linkedUser.allowedFacilities();
+        expect(userFacilities).toEqual(expect.arrayContaining(expectedCombinedFacilities));
+      });
+
       it('should return linked facilities from user_facilities table', async () => {
-        const userFacilities = await user.allowedFacilities();
-        expect(userFacilities).toStrictEqual(validUserFacilities);
+        const userFacilities = await linkedUser.allowedFacilities();
+        expect(userFacilities).toStrictEqual(linkedUserFacilities);
       });
-      it('should return empty array if no linked facilities', async () => {
-        const userFacilities = await noFacilityUser.allowedFacilities();
+
+      it('should return empty array if no linked facilities with setting on', async () => {
```

Comment:
```suggestion
      it('should return an empty array if there are no linked facilities when restrictUsersToFacilities is enabled', async () => {
```
Again these it descriptions are conventionally very sentencey